pkg/controllers/node/termination/suite_test.go

```diff
@@ -126,25 +142,20 @@ var _ = Describe("Termination", func() {
 			ExpectNotFound(ctx, env.Client, node, nodeClaim)
 		})
 		It("should not race if deleting nodes in parallel", func() {
-			const numNodes = 10
-			var nodes []*corev1.Node
-			for i := 0; i < numNodes; i++ {
-				node = test.Node(test.NodeOptions{
-					ObjectMeta: metav1.ObjectMeta{
-						Finalizers: []string{v1.TerminationFinalizer},
-					},
-				})
+			nodes := lo.Times(10, func(_ int) *corev1.Node {
+				return test.NodeClaimLinkedNode(nodeClaim)
+			})
+			for _, node := range nodes {
 				ExpectApplied(ctx, env.Client, node, nodeClaim)
 				Expect(env.Client.Delete(ctx, node)).To(Succeed())
-				node = ExpectNodeExists(ctx, env.Client, node.Name)
-				nodes = append(nodes, node)
+				*node = *ExpectNodeExists(ctx, env.Client, node.Name)
 			}
 
 			// Reconcile twice, once to set the NodeClaim to terminating, another to check the instance termination status (and delete the node).
 			for range 2 {
 				var wg sync.WaitGroup
 				// this is enough to trip the race detector
-				for i := 0; i < numNodes; i++ {
+				for i := range nodes {
```

Comment:
nit: These are changes that make the diff larger -- fine with them, just be careful with it for such a large PR already